File: endpoints/photos.py
from silence.decorators import endpoint

@endpoint(
    route="/photos",
    method="GET",
    sql="SELECT * FROM Photos WHERE visibility='public' ORDER BY date DESC"
)
def get_all():
    pass

###############################################################################

@endpoint(
    route="/photos/siguiendo/$userId",
    method="GET",
    sql="SELECT * FROM photos NATURAL JOIN seguidores WHERE seguidorId = $userId AND userId=seguidosId AND visibility='public' GROUP BY photoId ORDER BY date DESC"
)
def get_by_id():
    pass

###############################################################################

@endpoint(
    route="/photos/$photoId",
    method="GET",
    sql="SELECT * FROM Photos WHERE photoId = $photoId"
)
def get_by_id():
    pass

###############################################################################

@endpoint(
    route="/photos/media/$photoId",
    method="GET",
    sql="SELECT P.*, V.*, round(AVG(V.valor),1) AS media FROM photos P JOIN valoracion V ON (P.photoId=V.photoId)  WHERE V.photoId = $photoId"
)
def get_by_id():
    pass

###############################################################################

@endpoint(
    route="/photos/media",
    method="GET",
    sql="SELECT round(avg(valor),1) AS media,P.*,U.* FROM valoracion V JOIN photos P ON (V.photoId = P.photoId) JOIN users U ON (P.userId=U.userId) GROUP BY V.photoId order BY media desc"
)
def get_by_id():
    pass

###############################################################################

@endpoint(
    route="/photos/user/$userId",
    method="GET",
    sql="SELECT * FROM Photos WHERE userId = $userId AND visibility='public' ORDER BY date DESC"
)
def get_by_id():
    pass

###############################################################################

@endpoint(
    route="/photos/count/user/$userId",
    method="GET",
    sql="SELECT * FROM photos NATURAL JOIN users WHERE userId = $userId"
)
def get_by_id():
    pass

###############################################################################

@endpoint(
    route="/photos/userprivate/$userId",
    method="GET",
    sql="SELECT * FROM Photos WHERE userId = $userId AND visibility='private' ORDER BY date DESC"
)
def get_by_id():
    pass

###############################################################################

@endpoint(
    route="/photos",
    method="POST",
    sql="INSERT INTO Photos (title, description, url, visibility, userId) VALUES ($title, $description, $url, $visibility, $userId)",
    description="Creates a new photo",
    auth_required=True,
)
def create(title, description, url, visibility, userId):
    pass

###############################################################################

@endpoint(
    route="/photos/$photoId",
    method="PUT",
    sql="UPDATE Photos SET title = $title, description = $description, url = $url, visibility = $visibility WHERE photoId = $photoId",
    description="Updates an existing photo",
    auth_required=True,
)
def update(title, description, url, visibility):
    pass

###############################################################################

@endpoint(
    route="/photos/$photoId",
    method="DELETE",
    sql="DELETE FROM Photos WHERE photoId = $photoId",
    description="Removes a photo",
    auth_required=True,
)
def delete():
    pass

File: docs/js/api/photos.js
"use_strict";
import { BASE_URL, requestOptions } from "./common.js";
const photosAPI = {
    getAll: function () {
        return new Promise(function (resolve, reject) {
            axios
                .get(`${BASE_URL}/photos`, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },

    getSiguiendo: function (userId) {
        return new Promise(function (resolve, reject) {
            axios
                .get(`${BASE_URL}/photos/siguiendo/${userId} `, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },

    getMediaPhoto: function (photoId) {
        return new Promise(function (resolve, reject) {
            axios
                .get(`${BASE_URL}/photos/media/${photoId} `, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },

    getMediaPhotos: function () {
        return new Promise(function (resolve, reject) {
            axios
                .get(`${BASE_URL}/photos/media `, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },

    getById: function (photoId) {
        return new Promise(function (resolve, reject) {
            axios
                .get(`${BASE_URL}/photos/${photoId} `, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },

    getCountByUser: function (userId) {
        return new Promise(function (resolve, reject) {
            axios
                .get(`${BASE_URL}/photos/count/user/${userId} `, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },

    getPrivatesByUser: function (userId) {
        return new Promise(function (resolve, reject) {
            axios
                .get(`${BASE_URL}/photos/userprivate/${userId} `, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },

    getByUser: function (userId) {
        return new Promise(function (resolve, reject) {
            axios
                .get(`${BASE_URL}/photos/user/${userId} `, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },

    create: function(formData) {
        return new Promise(function(resolve, reject) {
            axios
                .post(`${BASE_URL}/photos`, formData, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },
    update: function(photoId, formData) {
        return new Promise(function(resolve, reject) {
            axios
                .put(`${BASE_URL}/photos/${photoId}`, formData,
                    requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },
    delete: function(photoId) {
        return new Promise(function(resolve, reject) {
            axios
                .delete(`${BASE_URL}/photos/${photoId} `, requestOptions
                )
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },
};
export { photosAPI };
